### PR TITLE
Feat/gravity

### PR DESCRIPTION
**Pull request by aaronw84**

### CODE REVIEW

File: TKGame/Character.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                vel.X = 3f;
+            // Move to left
+            else if (Keyboard.GetState().IsKeyDown(Keys.Left))
+                vel.X = -3f;
+            // No key pressed so don't move
+            else
+                vel.X = 0f;
+
+            // No double jumps due to hasJumped boolean
+            if (Keyboard.GetState().IsKeyDown(Keys.Space) && jumped == false)
```
Comment (NateCastleOIT) [inline TKGame/Character.cs:43]: Why do we need to wait until the play jumps? The player will be use collisions to stop from falling into the floor so we should code gravity to be constant acceleration like real life.

File: TKGame/Character.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            {
+                // Height and speed when jumping
+                pos.Y -= 10f;
+                pos.Y = -5f;
```
Comment (NateCastleOIT) [inline TKGame/Character.cs:47]: Double assignment?

File: TKGame/Character.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            if (jumped == true)
+            {
+                // multiplier for gravity
+                //float i = 1;
+                // gravity formula  
+                vel.Y += 0.15f/** i*/;
+            }
+
+            // set hasJumped to false when character touches ground
+            if (pos.X + texture.Height >= 450)
```
Comment (NateCastleOIT) [inline TKGame/Character.cs:61]: What is 450? The top of the floor? This doesn't seem scalable

File: TKGame/Character.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+{
+    /*internal*/ class Character
+    {
+        Texture2D texture;
+        Vector2 pos;
+        Vector2 vel;
+        bool jumped;
+
+        // Character constructor
+        public Character(Texture2D newTexture, Vector2 newPos)
```
Comment (NathanGr33n) [inline TKGame/Character.cs:20]: Your going to want to add some summary comments in the future to give in code explanations for your functions and variables

**Review by NathanGr33n — COMMENTED** [review]
Looks good to me, may want to add some summary comments later.

File: TKGame/Character.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        {
+            texture = newTexture;
+            pos = newPos;
+            jumped = true;
+        }
+
+        // Player character movement, gravity, jumping 
+        public void Update(GameTime gameTime)
+        {
+            pos += vel;
```
Comment (NateCastleOIT) [inline TKGame/Character.cs:30]: Is there a reason behind updating this at the beginning rather than the end of the update?

File: TKGame/Character.cs [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+            // Move to right
+            if (Keyboard.GetState().IsKeyDown(Keys.Right))
```
Comment (NateCastleOIT) [inline TKGame/Character.cs:33]: Movement seems out of scope to the "gravity" feature

File: TKGame/Character.cs [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+            if (Keyboard.GetState().IsKeyDown(Keys.Space) && jumped == false)
+            {
+                // Height and speed when jumping
+                pos.Y -= 10f;
+                pos.Y = -5f;  
+                jumped = true;
+            }
+
+            // character has jumped (gravity takes effect)
+            if (jumped == true)
```
Comment (NateCastleOIT) [inline TKGame/Character.cs:52]: We shouldn't lock the gravity to the character having jumped, doesn't make much sense if I walk off a cliff I'll just keep going straight instead of falling.

**Review by NateCastleOIT — CHANGES_REQUESTED** [review]
Needs some refactoring, see comments
